include/albatross/src/utils/variant_utils.hpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2019 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#ifndef ALBATROSS_UTILS_VARIANT_UTILS_HPP_
+#define ALBATROSS_UTILS_VARIANT_UTILS_HPP_
+
+namespace albatross {
+
+namespace details {
+template <typename X> struct ToVariantIdentity {};
+} // namespace details
+
+/*
+ * In this case X is a variant which is compatible so we need to
+ * convert it.
+ */
+template <typename... Ts, typename X>
+inline std::enable_if_t<is_sub_variant<X, variant<Ts...>>::value, void>
+set_variant(variant<Ts...> &to_set, const X &x) {
```

Comment:
I'm not totally sure what to do here.  On one hand I like the current syntax `set_variant(to, from)` since it preserves the order of `to = from`, but it does conflict with code style which would probably prefer `set_variant(from, &to)`.  Maybe renaming it `assign_to(from, &to)` would be a good option?  Thoughts?

Reply:
Have you typically followed one style over the other in albatross? I'd suggest to keep the code inside of a project/repo consistent over starting to follow the suggested style half way through and having inconsistent style.

Reply:
Ah yeah actually a quick grep reminded me that I do this `set_subset` which passes the `to` as a pointer.  I'll switch to that.